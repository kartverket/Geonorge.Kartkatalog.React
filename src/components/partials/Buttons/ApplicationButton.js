import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export class ApplicationButton extends Component {
    constructor(props) {
        super(props);
        this.state = {
        };
    }

    isApplication() {
        return this.props.searchResult.Type === "software"
    }

    render() {
        if (this.isApplication(this.props.searchResult.Type)) {
            if (this.props.searchResult.DistributionUrl) {
                let distributionUrl = this.props.searchResult.DistributionUrl
                let icon = <FontAwesomeIcon icon={['far', 'external-link-square']} key="icon" />
                let buttonClass = 'on';
                let textContent = React.createElement('span', { key: "textContent" }, 'Til nettside')

                let childElements = [icon, textContent];
                return React.createElement('a', { href: distributionUrl, className: buttonClass }, childElements);
            }
            else {
                let icon = <FontAwesomeIcon icon={['far', 'external-link-square']} key="icon" />
                let buttonClass = 'btn btn-sm disabled off'
                let textContent = React.createElement('span', { key: "textContent" }, 'ikke tilgjengelig')
                let childElements = [icon, textContent];
                return React.createElement('span', { className: buttonClass }, childElements);
            }
        }
            return null
    }
}

ApplicationButton.propTypes = {
    searchResult: PropTypes.object.isRequired
}

export default connect(null, null)(ApplicationButton);