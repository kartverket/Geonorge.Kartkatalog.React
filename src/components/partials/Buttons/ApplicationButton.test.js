import React from 'react';
import { shallow } from "enzyme";
import { ApplicationButton } from './ApplicationButton';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function setupApplicationWithDistributionUrl() {
    const searchResult = {
        Uuid: '1234',
        Type: 'software',
        DistributionUrl: 'test'
    }
    const props = {
        searchResult: searchResult
    }
    const wrapper = shallow(<ApplicationButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupApplicationWithoutDistributionUrl() {
    const searchResult = {
        Uuid: '1234',
        Type: 'software',
    }
    const props = {
        searchResult: searchResult
    }
    const wrapper = shallow(<ApplicationButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupIsDatasetType() {
    const searchResult = {
        Uuid: '1234',
        Type: 'dataset',
    }
    const props = {
        searchResult: searchResult
    }
    const wrapper = shallow(<ApplicationButton {...props} />)

    return {
        props,
        wrapper
    }
}


describe('ApplicationButton', () => {
    
    it('should render self', () => {
        const { wrapper } = setupApplicationWithDistributionUrl()
        expect(wrapper).toMatchSnapshot();
    })

    it('Application with DistributionUrl', () => {
        const { wrapper } = setupApplicationWithDistributionUrl()

        expect(wrapper.hasClass('on')).toBe(true)
        expect(wrapper.prop("href")).toBe('test')
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('Til nettside')
    })

    it('Application without DistributionUrl', () => {
        const { wrapper } = setupApplicationWithoutDistributionUrl()
        
        expect(wrapper.hasClass('off')).toBe(true)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('ikke tilgjengelig')
    })

    it('Is dataset type', () => {
        const { wrapper } = setupIsDatasetType()
        
        expect(wrapper.hasClass()).toBe(false)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon)).toHaveLength(0)        
        expect(wrapper.find("span")).toHaveLength(0)
    })
})
