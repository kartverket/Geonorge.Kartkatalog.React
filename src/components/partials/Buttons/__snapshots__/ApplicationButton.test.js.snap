// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationButton should render self 1`] = `ShallowWrapper {}`;
